parse.go

```diff
@@ -241,24 +262,39 @@ func NewParser(config Config, dests ...interface{}) (*Parser, error) {
 	return &p, nil
 }
 
-func cmdFromStruct(name string, dest path, t reflect.Type) (*command, error) {
+func (cmd *command) addCommand(dest path, t reflect.Type) error {
```

Comment:
I've renamed them to `parseFieldsFromStructPointer` / `parseFieldsFromStructOrStructPointer` instead of `addFields...` since the command itself is parsing the structure recursively and the field itself is only added from where the function is called.

I did add comments to all 3 parsing functions to describe their intent. That was a good idea.